Modules/Core/CKTransactionalComponentDataSourceState+RemoveAll.mm

```diff
@@ -32,17 +32,14 @@ @implementation CKTransactionalComponentDataSourceState (RemoveAll)
 - (CKTransactionalComponentDataSourceChangeset*)removeAllChangeset
 {
   NSMutableSet *indexPaths = [NSMutableSet set];
-  NSMutableIndexSet *sections = [NSMutableIndexSet indexSet];
-
   [self enumerateObjectsUsingBlock:^(CKTransactionalComponentDataSourceItem *_, NSIndexPath *indexPath, BOOL *stop) {
     [indexPaths addObject:indexPath];
-    [sections addIndex:indexPath.section];
   }];
 
   return [[CKTransactionalComponentDataSourceChangeset alloc]
           initWithUpdatedItems:nil
           removedItems:indexPaths
-          removedSections:sections
+          removedSections:[NSIndexSet indexSetWithIndexesInRange:NSMakeRange(0, self.numberOfSections)]
```

Comment:
[`-[CKTransactionalComponentDataSourceState enumerateObjectsUsingBlock:]`](
https://github.com/eyeem/componentkit/blob/master/ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceState.mm#L46-L56) enumerates the items for each section, so if there are no items then the section isn't represented.